tests/bluechi_test/container.py

```diff
@@ -136,13 +136,15 @@ def wait_for_unit_state_to_be(
             delay: float = 0.5) -> bool:
         latest_state = ""
 
-        if self._is_unit_in_state(unit_name, expected_state):
+        if self.get_unit_state(unit_name) == expected_state:
             return True
 
         start = time.time()
         while (time.time() - start) < timeout:
             time.sleep(delay)
-            if self._is_unit_in_state(unit_name, expected_state):
+            state = self.get_unit_state(unit_name)
+            LOGGER.info(f"Found latest state: {state}")
+            if state == expected_state:
                 return True
```

Comment:
Is there a reason to loose debug logging by not using private function `_is_unit_in_state()`? I've extracted the comparison along with logging to prevent code duplication and having detailed logging to to check status change progress

Reply:
Locally it was running fine, so I assumed the flakyness has been fixed by chance. This PR should still be in draft mode, I'll have to debug this - the log change was just for that.

Also, we log on info level, so the debug statements arent shown. Could we change this to use debug by default?

Reply:
You can change the default log level:
https://github.com/eclipse-bluechi/bluechi/tree/main/tests#changing-log-level

I intentionally changed the most important log messages to INFO, because on DEBUG there are quite a lot debug messages from podman-py, which makes the log hard to read. If you think the level in `_is_unit_in_state` should be on INFO, feel free to change it